src/app/notebook/open/open.component.ts

```diff
@@ -23,9 +25,20 @@ export class OpenComponent {
   async importNotebook(event: Event) {
     const input = event.target as HTMLInputElement;
     if (input.files && input.files.length > 0) {
-      const blocks = JSON.parse(await input.files[0].text()).blocks;
-      await this.manager.bulkInsertBlocks(blocks);
+      const currentBlocks = (await this.manager.exportNotebook(url.read("p"), url.read("t"))).blocks;
+      let currentMaxndex = currentBlocks.reduce((max, _, index) => Math.max(max, index), 0);
+      let newBlocks = JSON.parse(await input.files[0].text()).blocks;
+      newBlocks = newBlocks.map((block: BlockDocument) => {
+        if (block && block.index !== undefined) {
+          block.index = currentMaxndex;
+          currentMaxndex += 1;
+        }
+        return block;
+      }).filter((block: BlockDocument | undefined) => block !== undefined).forEach;
```

Comment:
Why did you do with the ```forEach```?

Reply:
Sorry, some of this is just me getting my thoughts together. Seems I missed that!